executor/index_merge_reader.go

```diff
@@ -860,6 +861,9 @@ func (e *IndexMergeReaderExecutor) getResultTask(ctx context.Context) (*indexMer
 
 func handleWorkerPanic(ctx context.Context, finished <-chan struct{}, ch chan<- *indexMergeTableTask, extraNotifyCh chan bool, worker string) func(r interface{}) {
 	return func(r interface{}) {
+		failpoint.Inject("testChangeProcessWorkerType", func() {
```

Comment:
I think this failpoint can be confusing.

Reply:
You can try following to reproduce this hang:
1. In `fetchLoopUnion` and `doIntersectionPerPartition`, first write `resultCh` then write `workCh`
2. After write `resultCh`, before write `workCh`, add a failpoint to call cancel func.

Reply:
> I think this failpoint can be confusing.

okk, I will add some comments.